fetch.bs

```diff
@@ -8481,6 +8613,77 @@ with a <var>promise</var>, <var>request</var>, <var>responseObject</var>, and an
 </div>
 
 
+<h3 id=fetch-later-method>FetchLater method</h3>
+
+<pre class=idl>
+
+dictionary DeferredRequestInit : RequestInit {
+  DOMHighResTimeStamp backgroundTimeout;
+};
+
+interface FetchLaterResult {
+  readonly attribute boolean sent;
+};
+
+partial interface mixin WindowOrWorkerGlobalScope {
+  [NewObject] FetchLaterResult fetchLater(RequestInfo input, optional DeferredRequestInit init = {});
+};
+</pre>
+
+<p>A {{FetchLaterResult}} has an associated <a>deferred fetch record</a>
+<dfn for=FetchLaterResult>deferred record</dfn>.
+
+<div algorithm>
+<p>The <dfn attribute for=FetchLaterResult><code>sent</code></dfn> getter steps are to return
+<a>this</a>'s <a for=FetchLaterResult>deferred record</a>'s <a for="deferred fetch record">sent</a>.
+</div>
+
+<div algorithm="dom-fetch-later">
+<p>The
+<dfn id=dom-global-fetch-later method for=WindowOrWorkerGlobalScope><code>fetchLater(<var>input</var>, <var>init</var>)</code></dfn>
+method steps are:
+
+<ol>
+ <li><p>Let <var>requestObject</var> be the result of invoking the initial value of {{Request}} as
+ constructor with <var>input</var> and <var>init</var> as arguments. If that threw an exception,
+ <a for=/>reject</a> <var>promise</var> with that exception and return <var>promise</var>.
+
+ <li><p>If <var>requestObject</var>'s <a for=Request>signal</a> is <a for=AbortSignal>aborted</a>,
+ then <a for=/>reject</a> <var>promise</var> with <var>requestObject</var>'s
+ <a for=Request>signal</a>'s <a for=AbortSignal>abort reason</a> and return <var>promise</var>.
+
+ <li><p>Let <var>request</var> be <var>requestObject</var>'s <a for=Request>request</a>.
+
+ <li><p>Let <var>backgroundTimeout</var> be null.
+
+ <li><p>If <var>init</var> is given and <var>init</var>["<code>backgroundTimeout</code>"]
+ <a for=map>exists</a> then set <var>backgroundTimeout</var> to
+ <var>init</var>["<code>backgroundTimeout</code>"].
+
+ <li><p>If <var>backgroundTimeout</var> is not a {{DOMHighResTimeStamp}} then throw a {{TypeError}}.
+
+ <li><p>Let <var>deferredRecord</var> be the result of calling
+ <a>request a deferred fetch</a> given <var>request</var> and <var>backgroundTimeout</var>. This
+ may throw an exception.
+
+ <li>
+  <p><a for=AbortSignal lt=add>Add the following abort steps</a> to <var>requestObject</var>'s
```

Comment:
My reading of this is that abort cannot be seen from JS. A request that has been aborted is indistinguishable from a request that is waiting to send. This is suprising to me and seems like a problem.

I feel like `pending` is a direct representation of the state. `sent==true` does not actually mean that it has been successfuly sent or even an attempt has been made yet. It also allows us to simplify the states of the beacon, merging `aborted` and `sent` into one.

That said, we might want to keep a distinction, in which case we would actually want to expose both bits.

Reply:
You can check `request.signal.aborted` for this.
Sent is specifically about whether it was scheduled & dispatched.

Reply:
I tend to agree, also `invoked`, though perhaps `committed`?
@annevk do you have a strong opinion about this?

Reply:
I think I'm happiest with `dispatched` or `committed`. `invoked` seems more like a callback and `scheduled` is kinda confusing because they've already scheduled it.

Reply:
@annevk friendly ping.

Reply:
Maybe `attempted`?

Reply:
I'm OK with `attempted`. @fergald @mingyc ?

Reply:
SGTM

Reply:
`attempted` implies that some action (e.g. a network request) has been attempted. That's not true. All that has happened is that we've gone past the point where this request can be cancelled. Under the hood, the renderer process has told the browser process to make the request. It could have done nothing at all or it could have completed it. `committed` or `dispatched` mean exactly that, `pending` means the opposite (but would differ from the other 2 when cancelled).

Reply:
Perhaps "triggered" would be a good word (or "started").

Reply:
either would be fine with me.